### ISSUE TITLE
no CUDA aware MPI in spack install

### ISSUE THREAD

**Issue by berceanu** [ComputationalRadiationPhysics/picongpu#2782]
Installing`PIConGPU` following the [spack recipe](https://picongpu.readthedocs.io/en/0.4.0/install/instructions/spack.html) with the default `backend=cuda` does not compile OpenMPI with CUDA support (see [package.py](https://github.com/ComputationalRadiationPhysics/spack-repo/blob/5f4de9d4dce9d85331edae6cedeeae43903b2c3b/packages/picongpu/package.py#L89) here).

**Comment by berceanu** [comment]
Should we check for CUDA-enabled MPI at runtime using `MPIX_Query_cuda_support()`, as described in the [manual](https://www.open-mpi.org/doc/v3.0/man3/MPIX_Query_cuda_support.3.php)?

**Comment by ax3l** [comment]
Yep, one does not necessarily need CUDA "awareness" in MPI, at least how we currently use it, as long as network drivers and CUDA don't steel each other's pinning: #438

Your #2777 `.tpl` file does have `--mca mpi_leave_pinned 0`, so this should hopefully be fine if I am not missing something... But that does not mean it's not an MPI (or MPI-IO) issue...

Anyway, back to this thread's topic: non-cuda aware MPI should be fine for now for our spack package. (Also, afaik the virtual `mpi` package in Spack does not yet support variants to enable it generally.)
